### ISSUE TITLE
`cabal new-test` fails if there is no tests

### ISSUE THREAD

**Issue by vrom911** [haskell/cabal#5435]
If my package has no test stanza and I run `cabal new-test` I get:

```
cabal: Cannot run tests for the target '' which refers to the package
mypackage-0.0.0 because it does not contain any test suites.
```
with exit code 1.

Though it would be much friendly if this command would succeed with exit code 0 but produce some descriptive message. For example, it would help to make a uniform config for CI (ex: Travis).

My version of `cabal`:
```
$ cabal --version
cabal-install version 2.3.0.0
compiled using version 2.3.0.0 of the Cabal library
```

**Comment by m-renaud** [comment]
Would hiding this behaviour behind a flag be an acceptable solution? I imagine that changing this behaviour for the default could very well break existing usages. I can see situations where both return values could be useful though, so it seems reasonable to support.

Maybe something like `--allow-no-test-suites` or something like that.

**Comment by vrom911** [comment]
@m-renaud, yes, I agree that this is a reasonable solution!

**Comment by fgaz** [comment]
@m-renaud I'd like the opposite: `--disallow-no-test-suites`, since the identity of test results is success

**Comment by m-renaud** [comment]
@fgaz I'm a slight -1 on that name since it's a double negative which is confusing to begin with. It would mean to trigger the new behaviour would need to pass `--disallow-no-test-suites=False` which is just mind bending ;P

Also, for optional flags the usual default value is False, but this one would require a default value of True which is also inconsistent.

**Comment by vrom911** [comment]
I also like this suggestion more as it makes more sense that "no tests" is the positive result of the tests. I'm just happy with any solution at the moment, cause I need to set up a bunch of `if`s for Cabal to overcome this issue on CI.

**Comment by fgaz** [comment]
Fixed by #5768
